domains/client/relayer/src/worker.rs

```diff
@@ -175,6 +177,16 @@ pub async fn start_relaying_messages<CClient, Client, CBlock, Block, SO>(
             continue;
         }
 
+        if let Some(ref snap_sync_orchestrator) = snap_sync_orchestrator {
+            if !snap_sync_orchestrator.domain_snap_sync_finished() {
+                tracing::debug!(
+                    target: LOG_TARGET,
+                    "Domain snap sync: skipping message processing..."
+                );
+                continue;
+            }
+        }
```

Comment:
The relayer will skip the block when the consensus chain is in major sync (we can change it to check if the domain chain is in major sync if that is supported) so this check may be unnecessary.